packages/netlify-cms-core/src/components/Editor/EditorToolbar.js

```diff
@@ -303,16 +303,25 @@ class EditorToolbar extends React.Component {
 
   renderWorkflowStatusControls = () => {
     const { isUpdatingStatus, onChangeStatus, currentStatus, t, useOpenAuthoring } = this.props;
+
+    let buttonText = '';
+
+    if (currentStatus === status.get('DRAFT')) {
+      buttonText = t('editor.editorToolbar.draft');
+    } else if (currentStatus === status.get('PENDING_REVIEW')) {
+      buttonText = t('editor.editorToolbar.inReview');
+    } else if ( currentStatus === status.get('PENDING_PUBLISH')) {
+      buttonText = t('editor.editorToolbar.ready');
+    } else {
+      buttonText = isUpdatingStatus ? t('editor.editorToolbar.updating') : t('editor.editorToolbar.setStatus');
```

Comment:
I believe this will be dead code after this PR. Have to managed to reach the block inside this `else`?

Also, `setStatus` can probably be removed